### PR TITLE
Adding the cheriot-audit tests to the test suite

### PR DESCRIPTION
**Pull request by matajoh**
This PR adds the test suite from https://github.com/CHERIoT-Platform/cheriot-audit/ to this repo.

### CODE REVIEW

File: tests/cheriot/builtins.cpp [hunk 1]
```diff
@@ -0,0 +1,181 @@
+#include "../rego_test.h"
```
Comment (mjp41) [inline tests/cheriot/builtins.cpp:1]: Has this code come from cheriot?  Do we need some attribution here?

File: tests/cheriot/builtins.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        return scalar(false);
+      }
+      string = string.substr(compartmentNameString.size());
+    }
+    if (!string.starts_with("_"))
+    {
+      return scalar(false);
+    }
+    string = string.substr(1);
+    // The way that rego-cpp exposes snmalloc can cause the realloc here to
```
Comment (mjp41) [inline tests/cheriot/builtins.cpp:53]: Aiming to fix this with https://github.com/microsoft/trieste/pull/117

**Review by mjp41 — COMMENTED** [review]
Anything in particular you would like me to look at in a review?

File: tests/cheriot/builtins.cpp (hunk 1 above)
Comment (davidchisnall) [reply tests/cheriot/builtins.cpp:1]: This is our SPDX header:

```
// Copyright SCI Semiconductor and CHERIoT Contributors.
// SPDX-License-Identifier: MIT
```

**Comment by matajoh** [comment]
> Anything in particular you would like me to look at in a review?

I'm talking with David about options. I think we're going to change the tactic to include his repo and and build it against the current repo/branch similar to how we do with Python and Rust, instead of manually copying it in as I've done. He needs to do a bit of instrumentation first, but that way we don't have duplication.

**Comment by davidchisnall** [comment]
CHERIoT-Platform/cheriot-audit#4 should make that option easier.

**Comment by matajoh** [comment]
This has been done via another means.

**Pull request closed without merging**